readthedocs/doc_builder/python_environments.py

```diff
@@ -511,6 +528,18 @@ def _update_conda_startup(self):
             cwd=self.checkout_path,
         )
 
+    def _install_mamba(self):
+        self.build_env.run(
+            'conda',
+            'install',
+            '--yes',
+            '--quiet',
+            '--name=base',
+            '--channel=conda-forge',
```

Comment:
@SylvainCorlay! is it possible to install `mamba` from a different, and smaller, channel than `conda-forge` here?

I can't use `micromamba` at this point for "reasons" and I would like to installing it with `conda` but ideally using a channel that only contains `mamba`, so it does not make conda to consume too many resources.

Reply:
Would you consider using a miniforge flavor including mamba instead of miniconda.

Reply:
Comment from the peanut gallery: https://github.com/regro/conda-metachannel/issues/31#issuecomment-738683086 conda-metachannel service is down, unfortunately.

Perhaps if there was a `mamba` channel, it would help the bootstrapping problem. `conda install mamba -c mamba` would be blazing fast, and from there mamba & conda-forge could be used as usual.

About mamba in miniforge... from what I read in https://github.com/conda-forge/miniforge/issues/23, looks like there is no consensus (the path of least resistance seems to be creating a `miniforge-mamba` or a `microforge` with `micromamba`?)

Reply:
I can't use anything different than regular `conda` (like `micromamba`, `miniforge`, etc) _at this moment_ because I can't modify the Docker image we are currently using (I will be able to do this in the future, but I don't know exactly _when_ yet).

So, given this current restriction, I was asking if something like @astrojuanlu mentioned already existed (`conda install mamba -c mamba`) because calling `conda install mamba -c conda-forge` currently has the same problem of consuming a lot of resources just because it uses `conda-forge` channel that contains millions  of packages. Although the problem is there, it's not a blocker to start testing `mamba` here, but having a better workaround for this would be good.

> Comment from the peanut gallery: [regro/conda-metachannel#31 (comment)](https://github.com/regro/conda-metachannel/issues/31#issuecomment-738683086) conda-metachannel service is down, unfortunately.

This was a good idea when @astrojuanlu commented it to me. However, if it's currently down it doesn't seem to be something we can rely on by default :disappointed:

Reply:
Gotcha

Reply:
My goal here is to avoid this random problem with the available tools and restrictions:

```
$ conda install --yes --quiet --name=base --channel=conda-forge mamba
Collecting package metadata (current_repodata.json): ...working... done
Solving environment: ...working... failed with initial frozen solve. Retrying with flexible solve.
Solving environment: ...working... failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): ...working... Killed


Command killed due to excessive memory consumption 
```

Reply:
FYI @humitos miniforge now includes a "mambaforge" installer which has mamba pre-installed.

https://github.com/conda-forge/miniforge

Reply:
I am looking into what we could do to allow `conda install mamba` from a raw miniconda to be faster.

Reply:
@humitos
One way to do this without having to maintain another channel with a copy of mamba and its dependencies is to use a conda lock file.

https://pypi.org/project/conda-lock/

You could generate a conda lock file offline, and use it in your script so that no solving is required.